backend/capellacollab/projects/toolmodels/modelsources/git/interface.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+
+import fastapi
+from fastapi import status
+from sqlalchemy import orm
+
+import capellacollab.projects.toolmodels.modelsources.git.models as git_models
+import capellacollab.settings.modelsources.git.crud as settings_git_crud
+import capellacollab.settings.modelsources.git.models as settings_git_models
+
+from . import interface_class
+from .github.interface import GithubInterface
+from .gitlab.interface import GitlabInterface
+
+
+def get_git_class(
+    db: orm.Session, git_model: git_models.DatabaseGitModel
+) -> interface_class.GitInterface:
+    git_instance = get_git_instance_for_git_model(db, git_model)
+    match git_instance.type:
+        case settings_git_models.GitType.GITLAB:
+            return GitlabInterface(git_model, git_instance)
+        case settings_git_models.GitType.GITHUB:
+            return GithubInterface(git_model, git_instance)
+        case _:
+            raise fastapi.HTTPException(
+                status_code=fastapi.status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail={
+                    "err_code": "GIT_INSTANCE_UNSUPPORTED",
+                    "reason": (
+                        "The used Git instance is neither a Gitlab nor a Github instance.",
+                    ),
+                },
+            )
```

Comment:
I wouldn't put this into a function but rather use a Factory for that and put it into `factory.py` in the `handler` subdirectory. The factory class could look like this (you may need to slightly adjust it)

```
class GitHandlerFactory:
    @staticmethod
    def create_git_handler(
        db: orm.Session, git_model: git_models.DatabaseGitModel
    ) -> models.GitHandler:
        git_instance = get_git_instance_for_git_model(db, git_model)
        if git_instance.type == settings_git_models.GitType.GITLAB:
            return gitlab_models.GitlabHandler(git_model, git_instance)
        elif git_instance.type == settings_git_models.GitType.GITHUB:
            return github_models.GithubHandler(git_model, git_instance)
        else:
            raise fastapi.HTTPException(
                status_code=fastapi.status.HTTP_500_INTERNAL_SERVER_ERROR,
                detail={
                    "err_code": "GIT_INSTANCE_UNSUPPORTED",
                    "reason": (
                        "The used Git instance is neither a Gitlab nor a Github instance.",
                    ),
                },
            )
```